ParkingZoneApp/Areas/Admin/Controllers/ParkingZoneController.cs

```diff
@@ -6,156 +6,139 @@
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.AspNetCore.Mvc.Rendering;
 using Microsoft.EntityFrameworkCore;
-using Microsoft.EntityFrameworkCore.Metadata.Internal;
+using Microsoft.Extensions.FileProviders;
 using ParkingZoneApp.Data;
 using ParkingZoneApp.Models;
+using ParkingZoneApp.Repositories;
 
-namespace ParkingZoneApp.Areas.Admin.Controllers
+namespace ParkingZoneApp.Areas.Admin
 {
     [Authorize]
     [Area("Admin")]
     public class ParkingZoneController : Controller
     {
-        private readonly ApplicationDbContext _context;
+        private readonly IRepository<ParkingZone> _repository;
```

Comment:
implement IParkingZoneRepository not IRepo. In order to use here you added IParkingZoneRepository